src/main/java/net/fabricmc/loom/task/LoomTasks.java

```diff
@@ -81,16 +81,21 @@ private static void registerIDETasks(TaskContainer tasks) {
 		});
 	}
 
-	private static void registerRunTasks(TaskContainer tasks) {
-		tasks.register("runClient", RunClientTask.class, t -> {
-			t.setDescription("Starts a development version of the Minecraft client.");
-			t.dependsOn("downloadAssets");
-			t.setGroup("fabric");
-		});
+	private static void registerRunTasks(TaskContainer tasks, Project project) {
+		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
+		extension.getRuns().create("client");
+		extension.getRuns().create("server", settings -> settings.programArg("nogui"));
 
-		tasks.register("runServer", RunServerTask.class, t -> {
-			t.setDescription("Starts a development version of the Minecraft server.");
-			t.setGroup("fabric");
+		project.afterEvaluate((p) -> {
+			for (LoomGradleExtension.RunConfigSettings config : extension.getRuns()) {
+				String configName = config.getName();
+				String taskName = "run" + configName.substring(0, 1).toUpperCase() + configName.substring(1);
+
+				tasks.register(taskName, RunGameTask.class, config).configure(t -> {
+					t.setDescription("Starts a development version of the Minecraft server.");
```

Comment:
This description will be off for a few. Should prob change it to something a little more generic.

Reply:
Whoops that was a copy paste mistake, I'll change it

Reply:
Ah good point, I forgot about that. Might just have been a copy paste mistake. It's fixed now